### PR TITLE
[Routing] i18n routing.

### PR DESCRIPTION
**Pull request by frankdejonge**

### CODE REVIEW

**Comment by frankdejonge** [comment]
I've chosen to use the PHP configurator DSL as the example. I hadn't seen it used previously in the examples. This might be something we'll need to look into in a broader sense.

**Comment by frankdejonge** [comment]
Status: Needs work

**Review by nicolas-grekas — APPROVED** [review]

**Comment by DylanDelobel** [comment]
If this one is merged, should close #9510

File: routing.rst [hunk 1]
```diff
@@ -136,6 +136,89 @@ use them later to :ref:`generate URLs <routing-generate>`.
...
+
+        use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+        use Symfony\Component\Routing\Annotation\Route;
+
+        class BlogController extends Controller
+        {
+            /**
+             * Matches /blog exactly
+             *
+             * @Route("/blog", name="blog_list")
```
Comment (xabbuh) [inline routing.rst]: The annotation example look wrong to me. It should have the same routes as the YAML, XML, and PHP examples.

File: routing.rst [hunk 2]
```diff
@@ -611,6 +694,18 @@ But if you pass extra ones, they will be added to the URI as a query string::
     ));
     // /blog/2?category=Symfony
 
+Generating Localized URLs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (xabbuh) [inline routing.rst]: should be stripped to match the length of the headline

File: routing.rst [hunk 3]
```diff
@@ -136,6 +136,89 @@ use them later to :ref:`generate URLs <routing-generate>`.
...
+        // config/routes.php
+        namespace Symfony\Component\Routing\Loader\Configurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('about_us', ['nl' => '/over-ons', 'en' => '/about-us'])
+                ->controller('App\Controller\CompanyController::about');
+        };
+
+When a localized route is matched Symfony will automatically know which locale
+should be used during the request. Defining routes this way also eliminated the
```
Comment (xabbuh) [inline routing.rst]: "eliminates"?

File: routing.rst [hunk 4]
```diff
@@ -611,6 +694,18 @@ But if you pass extra ones, they will be added to the URI as a query string::
     ));
     // /blog/2?category=Symfony
 
+Generating Localized URLs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you've defined localized routes Symfony will use the current request locale
```
Comment (xabbuh) [inline routing.rst]: [...] routes, Symfony [...]

File: routing.rst [hunk 5]
```diff
@@ -611,6 +694,18 @@ But if you pass extra ones, they will be added to the URI as a query string::
...
+as the default when generating routes. In order to generate the route for alternative
+locale you must pass the ``_locale`` in the parameters array::
```
Comment (xabbuh) [inline routing.rst]: "for alternative locales" or "for an alternative locale"

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Comment by michaelperrin** [comment]
This paragraph should be removed with this new documentation: http://symfony.com/doc/master/routing.html#translating-routes

File: routing.rst [hunk 6]
```diff
@@ -136,6 +136,89 @@ use them later to :ref:`generate URLs <routing-generate>`.
...
+
+        // config/routes.php
+        namespace Symfony\Component\Routing\Loader\Configurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('about_us', ['nl' => '/over-ons', 'en' => '/about-us'])
+                ->controller('App\Controller\CompanyController::about');
+        };
+
+When a localized route is matched Symfony will automatically know which locale
```
Comment (maxbeckers) [inline routing.rst]: `When a localized route is matched, Symfony will automatically know, which ...` and perhaps `matches` instead of `is matched`?

**Comment by javiereguiluz** [comment]
Frank, thanks for contributing these docs. Thank you also for implementing this fantastic feature! The community loves it and it's definitely one of the best new features of Symfony 4.1. By the way, congrats on your first Symfony Docs contribution too!

**Comment by frankdejonge** [comment]
@javiereguiluz thank you! 👍

**Pull request merged**